siliconcompiler/schema/schema_cfg.py

```diff
@@ -3506,6 +3484,177 @@ def schema_constraint(cfg):
+            switch="-server_nice <int>",
+            example= ["cli: -server_nice 100",
+                    "api: chip.set('server', 'nice', '100')"],
+            schelp="""
+            Specifies the scheduling priority for the job. Negative numbers
+            increases priority while positive numbers decrease priority.
+            For more information, see the job scheduler documentation.""")
+
+    scparam(cfg, ['server', 'timeout'],
+            sctype='str',
+            scope='job',
+            shorthelp="Server: Timeout value",
+            switch="-server_timeout <str>",
+            example= ["cli: -server_timeout 16:00",
+                    "api: chip.set('server', 'timeout', '16:00')"],
+            schelp="""
+            Specifies the limit on the total job run time.
+            Acceptable slurm time formats include "minutes", "minutes:seconds",
+            "hours:minutes:seconds", "days-hours", "days-hours:minutes",
+            and "days-hours:minutes:seconds".""")
+
+    scparam(cfg, ['server', 'interactive'],
+            sctype='bool',
+            scope='job',
+            shorthelp="Server: Request interactive session",
+            switch="-server_interactive <str>",
+            example= ["cli: -server_interactive true",
+                    "api: chip.set('server', 'interactive', True)"],
+            schelp="""
+            Specifies that job is interactive. In slurm, this would
+            translate to something like '-i --pty bash'.""")
+
+    scparam(cfg, ['server', 'defer'],
+            sctype='str',
+            scope='job',
+            shorthelp="Server: Start time",
+            switch="-server_defer <str>",
+            example= ["cli: -server_defer 16:00",
+                    "api: chip.set('server', 'defer', '16:00')"],
+            schelp="""
+            Defer initiation of job until the specified time. The parameter
+            is pass through string for remote job scheduler such as slurm.
+            For more information abotut the exact format specification, see
+            the job scheduler documentation. Examples of valid slurm specific
+            values include: now+1hour, 16:00, 010-01-20T12:34:00.""")
+
+    scparam(cfg, ['server', 'option'],
+            sctype='[str]',
+            shorthelp="Server: Job scheduler options",
+            switch="-server_option <str>",
+            example=[
+                "cli: -server_option '--pty bash'",
+                "api: chip.set('server','option, '--pty bash')"],
+            schelp="""
+            List of run time options passed through to the job scheduler for
+            expert users. See the job scheduler (slurm, lsf, gse) for
+            information about available options. The user specified options must
+            be compatible with the standardized job scheduler options (memory,
+            cores,...).""")
+
+    scparam(cfg, ['server', 'msgevent'],
+            sctype='str',
```

Comment:
The slurm job scheduler has specific keywords for handling different types of events and do not support lists. If we want to support a list it would need to be outside of slurm and we would need to come up with an enumerated list of keywords for those events.